.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,221 @@
+name: Build and push a Docker image
+
+on:
+  push:
+    branches:
+      - main
+      - stage
+      - run-integration-test
+    tags:
+      - '*'
+
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: 'ref to be deployed (e.g. "refs/heads/main", "v1.0.0", "2c0472cf")'
+        type: string
+        required: true
+        default: refs/heads/main
+env:
+  APP: bedrock
+  APPLICATION_REPOSITORY: mozilla/bedrock
+  IMAGE: bedrock
+  GAR_LOCATION: us
+  GCP_PROJECT_ID: moz-fx-bedrock-prod
+  GAR_REPOSITORY: bedrock
+  REF_ID: ${{ github.ref }}
+
+
+jobs:
+  build_and_publish_public_images:
+    name: Build public Bedrock images and push to Docker hub
+    runs-on: [self-hosted, nonprod, prod, us-central1, bedrock]
```

Comment:
Do we want this to run on both `nonprod` and `prod`? Or is there some case somewhere that puts the `prod` build on `prod` and `nonprod` on `nonprod`?